freqtrade/rpc/api_server/ws/channel.py

```diff
@@ -170,36 +190,29 @@ async def disconnect_all(self):
         Disconnect all Channels
         """
         with self._lock:
-            for websocket, channel in self.channels.copy().items():
-                if not channel.is_closed():
-                    await channel.close()
-
-            self.channels = dict()
+            for websocket in self.channels.copy().keys():
+                await self.on_disconnect(websocket)
 
-    async def broadcast(self, data):
+    async def broadcast(self, message: WSMessageSchema):
         """
-        Broadcast data on all Channels
+        Broadcast a message on all Channels
 
-        :param data: The data to send
+        :param message: The message to send
         """
         with self._lock:
-            message_type = data.get('type')
-            for websocket, channel in self.channels.copy().items():
-                if channel.subscribed_to(message_type):
-                    if not channel.queue.full():
-                        await channel.send(data)
-                    else:
-                        logger.info(f"Channel {channel} is too far behind, disconnecting")
-                        await self.on_disconnect(websocket)
+            for channel in self.channels.copy().values():
+                if channel.subscribed_to(message.type):
```

Comment:
hm - seems like the type we pass in here ain't of type `WSMessageSchema` - but of type dict.

```
2022-10-24 19:54:03,749 - freqtrade.rpc.api_server.webserver - ERROR - Exception happened in background task: 'dict' object has no attribute 'type'
Traceback (most recent call last):
  File "/home/xmatt/devel/cryptos/freqtrade/freqtrade/rpc/api_server/webserver.py", line 200, in _broadcast_queue_data
    await self._ws_channel_manager.broadcast(message)
  File "/home/xmatt/devel/cryptos/freqtrade/freqtrade/rpc/api_server/ws/channel.py", line 204, in broadcast
    if channel.subscribed_to(message.type):
AttributeError: 'dict' object has no attribute 'type'

```

Reply:
I have no idea how this slipped past me.

Reply:
it's the call to broadcast that's not typed (the queue has no type - so it's implicitly returning ANY)